crates/string_wizard/src/magic_string/mod.rs

```diff
@@ -195,13 +182,21 @@ impl<'text> MagicString<'text> {
       None
     } else {
       self.split_at(text_index);
-      // TODO: safety: using `unwrap_unchecked` is fine.
-      let idx = self.chunk_by_end.get(&text_index).unwrap();
-      Some(&mut self.chunks[*idx])
+      let idx = self.chunk_by_end.get(&text_index)?;
+      self.chunks.get_mut(*idx)
     }
   }
 }
 
+impl<'text> std::fmt::Display for MagicString<'text> {
```

Comment:
I think again. Does this create unnecessary `String`?

we could just write

```rust
self.fragments().try_for_each(|f| write!(f, "{ret}"));
```

But this code will lose the allocate-only-once optimization. I failed to find related API to do the same thing as `String::reserve` did.